File: app/src/main/java/fansirsqi/xposed/sesame/util/FansirsqiUtil.java
package fansirsqi.xposed.sesame.util;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import org.json.JSONObject;
public class FansirsqiUtil {
  // 定义一言API的URL
  private static final String HITOKOTO_API_URL = "https://v1.hitokoto.cn/";
  private static final ExecutorService executorService = Executors.newSingleThreadExecutor(); // 创建单线程执行器
  /**
   * 获取一言句子并格式化输出。
   *
   * @param callback 回调接口，用于返回获取到的句子
   */
  public static void getOneWord(final OneWordCallback callback) {
    executorService.execute(
        () -> {
          StringBuilder response = new StringBuilder();
          try {
            // 创建 URL 对象并打开连接
            HttpURLConnection connection = (HttpURLConnection) new URL(HITOKOTO_API_URL).openConnection();
            connection.setRequestMethod("GET");
            connection.setConnectTimeout(5000); // 设置连接超时时间
            connection.setReadTimeout(5000); // 设置读取超时时间
            // 读取响应数据
            try (BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
              String line;
              while ((line = reader.readLine()) != null) {
                response.append(line);
              }
            }
            // 解析 JSON 数据
            JSONObject jsonObject = new JSONObject(response.toString());
            String hitokoto = jsonObject.optString("hitokoto", "忘形雨笠烟蓑，知心牧唱樵歌。\n明月清风共我，闲人三个，从他今古消磨。");
            String from = jsonObject.optString("from", "天净沙·渔父");
            // 格式化输出句子
            String formattedSentence = String.format("%s\n\n                    -----Re: %s", hitokoto, from);
            callback.onSuccess(formattedSentence); // 调用回调返回成功结果
          } catch (Exception e) {
            String hitokoto = "忘形雨笠烟蓑，知心牧唱樵歌。\n明月清风共我，闲人三个，从他今古消磨。";
            String from = "天净沙·渔父";
            String formattedSentence = String.format("%s\n\n                    -----Re: %s", hitokoto, from);
            Log.printStackTrace(e);
            callback.onFailure(formattedSentence); // 调用回调返回失败信息
          }
        });
  }
  // 定义回调接口
  public interface OneWordCallback {
    void onSuccess(String result); // 成功回调
    void onFailure(String error); // 失败回调
  }
}

File: app/src/main/java/fansirsqi/xposed/sesame/util/FansirsqiUtil.kt
package fansirsqi.xposed.sesame.util

import android.os.Handler
import android.os.Looper
import org.json.JSONObject
import java.io.BufferedReader
import java.io.InputStreamReader
import java.lang.Exception
import java.net.HttpURLConnection
import java.net.URL
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors
import fansirsqi.xposed.sesame.util.GlobalThreadPools
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

object FansirsqiUtil {
    // 定义一言API的URL
    private const val HITOKOTO_API_URL = "https://v1.hitokoto.cn/"

    /**
     * 获取一言句子并格式化输出。
     *
     * @param callback 回调接口，用于返回获取到的句子
     */
    /**
     * 获取一言（挂起函数），推荐在协程中使用
     * @return 成功返回句子，失败返回默认句子
     */
    suspend fun getOneWord(): String = withContext(Dispatchers.IO) {
        return@withContext try {
            val connection = URL(HITOKOTO_API_URL).openConnection() as HttpURLConnection
            connection.requestMethod = "GET"
            connection.connectTimeout = 5000
            connection.readTimeout = 5000

            val response = BufferedReader(InputStreamReader(connection.inputStream)).use { reader ->
                reader.readText()
            }

            val jsonObject = JSONObject(response)
            val hitokoto = jsonObject.optString(
                "hitokoto",
                " 去年相送，余杭门外，飞雪似杨花。\n今年春尽，杨花似雪，犹不见还家。"
            )
            val from = jsonObject.optString("from", "少年游·润州作代人寄远 苏轼")

            "$hitokoto\n\n                    -----Re: $from"
        } catch (e: Exception) {
            Log.printStackTrace(e)
            " 去年相送，余杭门外，飞雪似杨花。\n今年春尽，杨花似雪，犹不见还家。\n\n                    -----Re: 少年游·润州作代人寄远 苏轼"
        }
    }

    /**
     * 生成随机字符串
     * @param length 字符串长度
     */
    fun getRandomString(length: Int): String {
        val charPool: List<Char> = ('a'..'z') + ('A'..'Z') + ('0'..'9')
        return (1..length)
            .map { kotlin.random.Random.nextInt(0, charPool.size) }
            .map(charPool::get)
            .joinToString("")
    }

}
